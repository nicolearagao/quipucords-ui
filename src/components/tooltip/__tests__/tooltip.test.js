import React from 'react';
import Tooltip from '../tooltip';

describe('Tooltip Component', () => {
  it('should render a basic component', () => {
    const props = {
      id: 'test',
      content: 'hello world'
    };

    const component = renderComponent(<Tooltip {...props}>Test tooltip</Tooltip>);
    expect(component).toMatchSnapshot('basic');
  });

  it('should render a popover', () => {
    const props = {
      id: 'test',
      isPopover: true,
      content: 'hello world'
    };

    const component = renderComponent(<Tooltip {...props}>Test popover</Tooltip>);
    expect(component).toMatchSnapshot('popover');
  });
});
