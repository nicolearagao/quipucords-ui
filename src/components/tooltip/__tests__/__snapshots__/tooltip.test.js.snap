// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Component should render a basic component: basic 1`] = `
<tooltip
  props="{
  "id": "test",
  "content": "hello world",
  "children": [
    {}
  ],
  "isPopover": false,
  "placement": "top",
  "delayShow": 500
}"
>
  <div
    class="quipucords-tooltip__wrapper"
  >
    Test tooltip
  </div>
</tooltip>
`;

exports[`Tooltip Component should render a popover: popover 1`] = `
<tooltip
  props="{
  "id": "test",
  "isPopover": true,
  "content": "hello world",
  "children": [
    {}
  ],
  "placement": "top",
  "delayShow": 500
}"
>
  <div
    class="quipucords-popover__wrapper"
  >
    Test popover
  </div>
</tooltip>
`;
